closed/src/java.base/share/classes/com/sun/crypto/provider/NativeGaloisCounterMode.java

```diff
@@ -437,13 +461,15 @@ int encryptFinal(ByteBuffer src, ByteBuffer dst)
      * @return the number of bytes placed into the <code>out</code> buffer
      */
     int decrypt(byte[] in, int inOfs, int len, byte[] out, int outOfs) {
-        checkDataLength(ibuffer.size(), len);
-
-        if (len > 0) {
-            // store internally until decryptFinal is called because
-            // spec mentioned that only return recovered data after tag
-            // is successfully verified
-            ibuffer.write(in, inOfs, len);
+        synchronized (this) {
+            checkDataLength(ibuffer.size(), len);
```

Comment:
Possible `NullPointerException` here; please do as in `encrypt`:
```
    if (ibuffer == null) {
        ibuffer = new ByteArrayOutputStream();
    }
```

Reply:
Done.